edx_exams/apps/core/management/commands/bulk_add_course_staff.py

```diff
@@ -63,32 +63,26 @@ def add_course_staff_from_csv(self, csv_file, batch_size, batch_delay):
         Add the given set of course staff provided in csv
         """
         reader = list(csv.DictReader(csv_file))
-        users_to_create = []
-        users_existing = {u.username for u in User.objects.filter(username__in=[r.get('username') for r in reader])}
-        for row in reader:
-            if row.get('username') not in users_existing:
-                users_to_create.append(row)
-                users_existing.add(row.get('username'))
 
         # bulk create users
-        for i in range(0, len(users_to_create), batch_size):
+        for i in range(0, len(reader), batch_size):
             User.objects.bulk_create(
-                User(
-                    username=user.get('username'),
-                    email=user.get('email'),
-                )
-                for user in users_to_create[i:i + batch_size]
+                (User(
+                    username=row.get('username'),
+                    email=row.get('email'),
+                ) for row in reader[i:i + batch_size]),
+                ignore_conflicts=True,
             )
             time.sleep(batch_delay)
 
         # bulk create course staff
         for i in range(0, len(reader), batch_size):
```

Comment:
NIT: Is there any reason we need to do this in two separate loops? Works either way so I'd leave it up to you on if you think consolidating makes sense or no.

Reply:
You're right. I had two separate loops when we were creating an "existing_users" list to iterate on, but now that we're using ignore_conflics and iterating on the same list, I can remove before merging.